### PR TITLE
CRUD Operations for Driver Shift table

### PR DESCRIPTION
**Pull request by cjyu81**
In this branch there is:

- Missing CRUD operation Backend endpoints for Update and Delete are added for Shift Table
- IndexPage page component
- EditPage page component
- Deletion of apparently-denigrated ShiftPage component

All associated tests and stories are implemented and work 100%. The branch is large because I thought the elements would depend on each other(I had an error working on IndexPage, causing me to work on the missing CRUD operations and finishing EditPage just in case its absence was causing an error).
Admin users have total read-write access.
Non-admin users have only read access.

### CODE REVIEW

**Review by Mohamed-Elfouly — APPROVED** [review]
LGTM

**Review by Mohamed-Elfouly — APPROVED** [review]
Fixed Merge conflict, same as last time

File: frontend/src/main/pages/Shift/ShiftIndexPage.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    //     <BasicLayout>
+    //     <div className="pt-2">
+    //         <Button style={{ float: "right" }} as={Link} to="/shift/create">
+    //         Create Shift
+    //         </Button>
+    //         <h1>Shifts</h1>
+    //         <ShiftTable shift={shifts} currentUser={currentUser} />
+    //     </div>
+    //     </BasicLayout>
+    // )
```
Comment (pconrad) [inline frontend/src/main/pages/Shift/ShiftIndexPage.js:55]: Oh for heavens' sake.   Really?
```suggestion
```

File: frontend/src/tests/components/Shift/ShiftTable.test.js [hunk 2]
```diff
@@ -41,7 +56,7 @@ describe("ShiftTable tests", () => {
 
         expectedFields.forEach( (field)=> {
           const header = getByTestId(`${testId}-cell-row-0-col-${field}`);
-          expect(header).toBeInTheDocument();
+          expect(header).toBeInTheDocument();//test failing
```
Comment (pconrad) [inline frontend/src/tests/components/Shift/ShiftTable.test.js:59]: Sigh

```suggestion
          expect(header).toBeInTheDocument(); 
```

**Pull request merged**